gettsim/policy_environment.py

```diff
@@ -358,9 +361,55 @@ def _load_parameter_group_from_yaml(date, group, parameters=None):
 
     tax_data["datum"] = date
 
+    # Load rounding parameters if they exist
+    if "rounding" in raw_group_data:
+        tax_data["rounding"] = _load_rounding_parameters(
+            date, raw_group_data["rounding"]
+        )
     return tax_data
 
 
+def _load_rounding_parameters(date, rounding_spec):
+    """Load rounding parameters for a specific date from a dictionary.
+
+    Parameters
+    ----------
+    date : datetime.date
+        The date for which the policy system is set up.
+    rounding_spec : dictionary
+        The dictionary that contains rounding parameters
+        for several functions and several dates
+
+    Returns:
+        dictionary: Rounding parameters for the specified date
+    """
+    out = {}
+    rounding_parameters = ["direction", "base"]
+
+    # Load values of all parameters at the specified date.
+    for function_name in rounding_spec.keys():
+
+        # Find all specified policy dates before date.
+        policy_dates_before_date = sorted(
+            key
+            for key in rounding_spec[function_name].keys()
+            if isinstance(key, datetime.date) and key <= date
+        )
+
+        # If any rounding specs are defined for a date before the specified
+        # date, copy them to params dictionary.
+        # If not, no key for this function name is created (this will
```

Comment:
I don't understand the second part ("If not, ...")

Reply:
```

        # If no appropriate rounding specs are found for the requested date, 
        # no key for this function name is created (this will raise an error 
        # later if the user doesn't adjust params df before calling 
        # `compute_taxes_and_transfers`).

Reply:
```
        # If no appropriate rounding specs are found for the requested date, 
        # the function will not appear in the returned dictionary. 
        # Note this will raise an error later unless the user adds an appropriate
        # rounding specification to the parameters... 
```

*(not sure about the wording in the very last bit, but we should be precise there. I don't think there is a "params df", for example?)*

Reply:
Sounds good: `... to the parameters dictionary.`